src/value.rs

```diff
@@ -77,40 +84,59 @@ impl Value {
     }
   }
 
-  /// Converts a value into a JSON string for distribution.
-  pub fn to_json(&self) -> String {
+  /// Creates a `Value` from a JSON string.
+  pub fn from_json(json: &str) -> Result<Value, Error> {
+    serde_json::from_str(json).map_err(Error::from)
+  }
+
+  /// Converts a `Value` into a JSON string.
+  pub fn to_json(&self) -> Result<String, Error> {
+    serde_json::to_string(self).map_err(Error::from)
+  }
+
+  /// Converts a `Value` into a nice and indented JSON string.
+  pub fn to_json_pretty(&self) -> Result<String, Error> {
+    serde_json::to_string_pretty(self).map_err(Error::from)
+  }
+}
+
+impl Serialize for Value {
+  #[inline]
+  fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error> where S: Serializer {
     match *self {
-      Value::Null => "null".to_owned(),
-      Value::Boolean(value) => if value { "true".to_owned() } else { "false".to_owned() },
-      Value::I64(value) => value.to_string(),
-      Value::F64(value) => value.to_string(),
-      Value::String(ref value) => "\"".to_owned() + &escape_string_for_json(value) + "\"",
-      Value::Object(ref map) => {
-        let mut json = "{".to_owned();
-        for (key, value) in map {
-          json.push_str("\"");
-          json.push_str(&escape_string_for_json(key));
-          json.push_str("\":");
-          json.push_str(&value.to_json());
-          json.push_str(",");
-        }
-        // Remove the last comma.
-        json.pop();
-        json.push_str("}");
-        json
-      },
-      Value::Array(ref vec) => {
-        let mut json = "[".to_owned();
-        for item in vec {
-          json.push_str(&item.to_json());
-          json.push_str(",");
-        }
-        // Remove the last comma.
-        json.pop();
-        json.push_str("]");
-        json
-      }
+      Value::Null => serializer.serialize_unit(),
+      Value::Boolean(value) => serializer.serialize_bool(value),
+      Value::I64(value) => serializer.serialize_i64(value),
+      Value::F64(value) => serializer.serialize_f64(value),
+      Value::String(ref value) => serializer.serialize_str(&value),
+      Value::Array(ref value) => value.serialize(serializer),
+      Value::Object(ref value) => value.serialize(serializer)
+    }
+  }
+}
+
+impl Deserialize for Value {
```

Comment:
No, not `impl for Value`. `de` `impl`s `Deserialize` for the structs and stuff defined in the `schema` module.